src/libcore/pin.rs

```diff
@@ -16,29 +16,30 @@
 //! but doesn't allow moving `T`. The pointer value itself (the `Box`) can still be moved,
```

Comment:
I think we're missing a crucial first paragraph here (or maybe the first paragraph itself just needs to be expanded). Something along the lines of:

> `Pin` ensures that a certain value has a stable address in memory, because other parts of the code relies on knowing that address. Specifically, when we say that a value is "pinned", we are making a promise that the pinned value's address cannot change until that value is dropped.

Reply:
I have rewritten the module intro, please have a look.